src/mpl/include/mpl_dbg.h

```diff
@@ -70,7 +70,11 @@
     {                                                                   \
         if ((_class & MPL_dbg_active_classes) && MPL_DBG_##_level <= MPL_dbg_max_level) { \
             char _s[MPL_DBG_MAXLINE];                                   \
-            MPL_snprintf _fmatargs ;                                    \
+            int _ret = MPL_snprintf _fmatargs ;                          \
+            /* by checking _ret, we supress -Wformat-trunction in gcc-8 */ \
+            if (_ret < 0) {                                              \
```

Comment:
A more typical pattern in MPL is to use `assert(cond)` instead of this check then abort style, but the fix itself looks good.

Reply:
I have changed it to `assert` and it seems work.